core/arch/arch_exports.h

```diff
@@ -1674,8 +1674,23 @@ _dynamorio_runtime_resolve(void);
 #    define APP_PARAM(mc, offs) APP_PARAM_##offs(mc)
 #endif /* X86/ARM */
 
-#define MCXT_SYSNUM_REG(mc) ((mc)->IF_X86_ELSE(xax, IF_ARM_ELSE(r7, r8)))
-#define MCXT_FIRST_REG_FIELD(mc) ((mc)->IF_X86_ELSE(xdi, r0))
+#if !(defined(MACOS) && defined(AARCH64))
+#    define MCXT_SYSNUM_REG(mc) ((mc)->IF_X86_ELSE(xax, IF_ARM_ELSE(r7, r8)))
+#    define MCXT_SET_SYSNUM_REG(mc, sysnr) \
+        ((mc)->IF_X86_ELSE(xax, IF_ARM_ELSE(r7, r8))) = (sysnr)
+#    define MCXT_FIRST_REG_FIELD(mc) &((mc)->IF_X86_ELSE(xdi, r0))
```

Comment:
As noted above it seems unnecessary to change this to take the address; that also seems misleading as the name does not sound like an address.